src/DynamicData.Tests/Cache/GroupOnDynamicFixture.cs

```diff
@@ -0,0 +1,313 @@
+﻿using System;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+
+using Bogus;
+using DynamicData.Tests.Domain;
+using DynamicData.Binding;
+using DynamicData.Kernel;
+using FluentAssertions;
+using Xunit;
+
+using Person = DynamicData.Tests.Domain.Person;
+using System.Reactive.Subjects;
+using System.Reactive;
+
+namespace DynamicData.Tests.Cache;
+
+public class GroupOnDynamicFixture : IDisposable
```

Comment:
Should probably add tests for Refresh and Move changesets, for completeness, whatever the expected behavior is.

Reply:
Also, since this is an operating receiving multiple observable inputs, there should be thread-safety testing.

Reply:
> Should probably add tests for Refresh and Move changesets, for completeness, whatever the expected behavior is.

Move operations are not supported.  I guess they could just be forwarded.

Refresh testing is a good idea.

Reply:
I added tests for Refresh events and some multithreaded stress tests that will fire all 3 observables independently from different threads a bunch of times and then verifies the results at the end.

Reply:
> Move operations are not supported.

Right, so the expected behavior should be that the operator ignores `Move` changes, for completeness' sake we should test that explicitly.

> fire all 3 observables independently from different threads a bunch of times and then verifies the results at the end

Sounds perfect.